### PR TITLE
Feature: Add SSL support to ZIO Http Client

### LINKED ISSUE #229: Unable to use zio-http as simple http client [Support request]
I’m trying to use zio-http as simple http client library with the SimpleClient example.

When I run the following code : 
```scala
object SimpleClient extends App {
  val env = ChannelFactory.auto ++ EventLoopGroup.auto()

  val program = for {
    _ <- console.putStrLn("Start")
    res <- Client.request("https://api.github.com/users/zio/repos")
    _ <- console.putStrLn("Finish")
    _ <- console.putStrLn {
      res.content match {
        case HttpData.CompleteData(data) => data.map(_.toChar).mkString
        case HttpData.StreamData(_)      => "<Chunked>"
        case HttpData.Empty              => ""
      }
    }
  } yield ()

  override def run(args: List[String]): URIO[zio.ZEnv, ExitCode] = program.exitCode.provideCustomLayer(env)

}
```

Current Behavior:  When I run the sample code the request is not executed and the program is frozen
```
sbt:hello> run
[info] running SimpleClient 
Start
```

Expected Behavior:  When I run the sample code *res.content* is displayed

Sample code : https://github.com/jlcanela/zhttp-client

Any help welcome, I’m 99% sure I did something stupid but cannot find it at the moment.

### PR DESCRIPTION
**Pull request by ShrutiVerma97**
fixes #229

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/service/client/ClientChannelInitializer.scala [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
 
-final case class ClientChannelInitializer[R](channelHandler: JChannelHandler) extends JChannelInitializer[JChannel]() {
+final case class ClientChannelInitializer[R](channelHandler: JChannelHandler, port: Int)
+    extends JChannelInitializer[JChannel]() {
   override def initChannel(ch: JChannel): Unit = {
-    ch.pipeline()
+    import io.netty.handler.ssl.SslContextBuilder
+    import io.netty.handler.ssl.util.InsecureTrustManagerFactory
+    val sslCtx: SslContext =
+      if (port == 443) SslContextBuilder.forClient().trustManager(InsecureTrustManagerFactory.INSTANCE).build()
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/service/client/ClientChannelInitializer.scala:14]: `InsecureTrustManagerFactory.INSTANCE ` should not be used use proper
 trustManager

File: zio-http/src/main/scala/zhttp/service/client/ClientChannelInitializer.scala [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+      else null
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/service/client/ClientChannelInitializer.scala:15]: use option

File: zio-http/src/main/scala/zhttp/service/client/ClientChannelInitializer.scala [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
 import io.netty.handler.codec.http.{HttpClientCodec => JHttpClientCodec}
+import io.netty.handler.ssl.SslContext
 import zhttp.core.{JChannel, JChannelHandler, JChannelInitializer, JHttpObjectAggregator}
 
-final case class ClientChannelInitializer[R](channelHandler: JChannelHandler) extends JChannelInitializer[JChannel]() {
+final case class ClientChannelInitializer[R](channelHandler: JChannelHandler, port: Int)
+    extends JChannelInitializer[JChannel]() {
   override def initChannel(ch: JChannel): Unit = {
-    ch.pipeline()
+    val sslCtx: Option[SslContext] = if (port == 443) Some(ClientSSLContext.getSSLContext()) else None
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/client/ClientChannelInitializer.scala:11]: - `port = 443` will not guarantee the request to be HTTPS
- Unnecessary creation of `Option` is going to be expensive.

File: zio-http/src/main/scala/zhttp/service/client/ClientSSLContext.scala [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+case object ClientSSLContext {
+  def getSSLContext(): SslContext = {
+
+    import io.netty.handler.ssl.SslContextBuilder
+    // truststore
+    val trustStore: Option[KeyStore]       = Option(KeyStore.getInstance("JKS"))
+    val trustStorePath: Option[String]     = Option(System.getProperty("javax.net.ssl.trustStore"))
+    val trustStorePassword: Option[String] = Option(System.getProperty("javax.net.ssl.trustStorePassword"))
+    val trustManagerFactory                = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm())
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/client/ClientSSLContext.scala:18]: Take input from user.

Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/client/ClientSSLContext.scala:18]: Reduce overall allocations while doing SSL requests.

File: zio-http/src/main/scala/zhttp/service/client/ClientChannelInitializer.scala [hunk 5]
```diff
@@ -1,14 +1,23 @@
 package zhttp.service.client
 
+import io.netty.channel.ChannelPipeline
 import io.netty.handler.codec.http.{HttpClientCodec => JHttpClientCodec}
+import io.netty.handler.ssl.SslContext
 import zhttp.core.{JChannel, JChannelHandler, JChannelInitializer, JHttpObjectAggregator}
 
-final case class ClientChannelInitializer[R](channelHandler: JChannelHandler) extends JChannelInitializer[JChannel]() {
+final case class ClientChannelInitializer[R](channelHandler: JChannelHandler, port: Int)
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/client/ClientChannelInitializer.scala:8]: Add unit test for an SSL request.

File: example/src/main/scala/SimpleClient.scala [hunk 6]
```diff
@@ -1,14 +1,19 @@
...
   val headers = List(Header.host("sports.api.decathlon.com"))
 
+  //Configuring Truststore for https(optional)
+  val trustStorePath                     = System.getProperty("javax.net.ssl.trustStore")
+  val trustStorePassword                 = System.getProperty("javax.net.ssl.trustStorePassword")
+  val trustStoreConfig: TrustStoreConfig = TrustStoreConfig(trustStorePath, trustStorePassword)
+
   val program = for {
-    res <- Client.request(url, headers)
+    res <- Client.request(url, headers, trustStoreConfig)
```
Comment (tusharmath) [inline example/src/main/scala/SimpleClient.scala:16]: Create a separate example for `SSL` update readme with example links.

File: project/BuildHelper.scala [hunk 7]
```diff
@@ -84,5 +88,6 @@ object BuildHelper extends ScalaSettings {
     Test / parallelExecution := true,
     incOptions ~= (_.withLogRecompileOnMacro(false)),
     autoAPIMappings := true,
+    javaOptions ++= javaxOptions
```
Comment (tusharmath) [inline project/BuildHelper.scala:91]: These settings are only used by the `example` project. This way it's being added globally.

**Comment by ShrutiVerma97** [comment]
closing in favor of #256

**Pull request closed without merging**